src/main/java/nl/tudelft/ti2206/group9/util/CollisionMap.java

```diff
@@ -0,0 +1,233 @@
+                            final Class<? extends AbstractEntity> collidee,
+                            final CollisionHandler<?, ?> handler) {
+        if (!handlers.containsKey(collider)) {
+            handlers.put(
+                    collider,
+                    new HashMap<>());
+        }
+
+        Map<Class<? extends AbstractEntity>, CollisionHandler<?, ?>
+                > map = handlers
+                .get(collider);
+        map.put(collidee, handler);
+    }
+
+    /**
+     * Handles the collision between two colliding parties, if a suitable
+     * collision handler is listed.
+     *
+     * @param <C1> The collider type.
+     * @param <C2> The collidee (AbstractEntity that was moved into) type.
+     * @param collider The collider.
+     * @param collidee The collidee.
+     */
+    public <C1 extends AbstractEntity, C2 extends AbstractEntity> void
+    collide(final C1 collider, final C2 collidee) {
+        Class<? extends AbstractEntity> colliderKey
+                = getMostSpecificClass(handlers, collider.getClass());
+        if (colliderKey == null) {
+            return;
+        }
+
+        Map<Class<? extends AbstractEntity>, CollisionHandler<?, ?>> map
+                = handlers.get(colliderKey);
+        Class<? extends AbstractEntity> collideeKey = getMostSpecificClass(map,
+                collidee.getClass());
+        if (collideeKey == null) {
+            return;
+        }
+
+        CollisionHandler<C1, C2> collisionHandler
+                = (CollisionHandler<C1, C2>) map
+                .get(collideeKey);
+        if (collisionHandler == null) {
+            return;
+        }
+
+        collisionHandler.handleCollision(collider, collidee);
+    }
+
+    /**
+     * Figures out the most specific class that is listed in the map. I.e. if A
+     * extends B and B is listed while requesting A, then B will be returned.
+     *
+     * @param map The map with the key collection to find a matching class in.
+     * @param key The class to search the most suitable key for.
+     * @return The most specific class from the key collection.
+     */
+    private Class<? extends AbstractEntity> getMostSpecificClass(
+            final Map<Class<? extends AbstractEntity>, ?> map,
+            final Class<? extends AbstractEntity> key) {
+        List<Class<? extends AbstractEntity>> collideeInheritance
+                = getInheritance(key);
+        for (Class<? extends AbstractEntity> pointer : collideeInheritance) {
+            if (map.containsKey(pointer)) {
+                return pointer;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Returns a list of all classes and interfaces the class inherits.
+     *
+     * @param clazz The class to create a list of super classes and
+     *              interfaces for.
+     * @return A list of all classes and interfaces the class inherits.
+     */
+    @SuppressWarnings("unchecked")
```

Comment:
Can you document (i.e. putting a comment in the code) why this SuppressWarnings is here?

<review,configuration